src/Security/SecAccessControl.cs

```diff
@@ -98,56 +100,19 @@ public enum SecAccessControlCreateFlags : long {
 #if !NET
 	[Mac (10,10)][iOS (8,0)]
 #endif
-	public partial class SecAccessControl : INativeObject, IDisposable {
-
-		private IntPtr handle;
-
-		public IntPtr Handle {
-			get {
-#if !COREBUILD
-				if (handle == IntPtr.Zero) {
-					IntPtr error;
-					handle = SecAccessControlCreateWithFlags (IntPtr.Zero, KeysAccessible.FromSecAccessible (Accessible), (nint)(int)Flags, out error);
-				}
-#endif
-				return handle;
-			}
-			internal set { handle = value; }
-		}
-
-		public void Dispose ()
-		{
-#if !COREBUILD
-			Dispose (true);
-#endif
-			GC.SuppressFinalize (this);
-		}
-
+	public partial class SecAccessControl : NativeObject {
 #if !COREBUILD
-		internal SecAccessControl (IntPtr handle)
+		internal SecAccessControl (IntPtr handle, bool owns)
+			: base (handle, owns)
 		{
-			// note: the properties won't match reality
-			Handle = handle;
 		}
 
 		public SecAccessControl (SecAccessible accessible, SecAccessControlCreateFlags flags = SecAccessControlCreateFlags.UserPresence)
+			: base (SecAccessControlCreateWithFlags (IntPtr.Zero, KeysAccessible.FromSecAccessible (accessible), (nint) (long) flags, out var _), true)
```

Comment:
Interesting, is this double type casting? `(nint) (long) flags`

Reply:
Yes, it's double type casting. `flags` is an enum, and we can't cast directly from an enum to `nint`, so we have to go through `long` first (which is the underlying type of the enum).